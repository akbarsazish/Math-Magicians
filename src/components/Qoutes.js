import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=education',
          {
            headers: {
              'X-Api-Key': 'RnsRXQB+C1lXlbhnahF5Rw==3ZjEYCzgkEmEnVdP',
            },
          },
        );

        if (!response.ok) {
          throw new Error('Network response was not ok');
        }

        const data = await response.json();
        setQuote(data[0]);
        setLoading(false);
      } catch (error) {
        setError('Failed to fetch quote.');
        setLoading(false);
      }
    };

    fetchQuote();
  }, []);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    /* eslint-disable */
    return <div>Error: {error}</div>;
  }
  return (
    <div className="quote">
      <p className="qouteText"> {quote.quote} - {quote.author}   </p>
    </div>
  );
};

export default Quote;
