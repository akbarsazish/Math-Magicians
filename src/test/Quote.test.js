import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Qoutes from '../components/Qoutes';

test('Qoutes component displays the loading state', async () => {
  render(<Qoutes />);
  const loadingElement = screen.getByText('Loading...');
  expect(loadingElement).toBeInTheDocument();

  // Using Jest snapshot for the loading state
  expect(screen.getByTestId('loadingState')).toMatchSnapshot();
});

test('Qoutes component displays an error message on fetch failure', async () => {
  jest.spyOn(global, 'fetch').mockRejectedValueOnce(new Error('Failed to fetch'));
  render(<Qoutes />);
  await waitFor(() => {
    screen.getByText('Error: Failed to fetch quote.');
  });

  // Using Jest snapshot for the error state
  expect(screen.getByTestId('errorState')).toMatchSnapshot();
});

test('Qoutes component displays the fetched quote correctly', async () => {
  const mockedData = {
    quote: 'Mocked quote text',
    author: 'Mocked author',
  };
  jest.spyOn(global, 'fetch').mockResolvedValueOnce({
    json: async () => [mockedData],
    ok: true,
  });
  render(<Qoutes />);
  await waitFor(() => {
    screen.getByText('Mocked quote text - Mocked author');
  });

  // Using Jest snapshot for the fetched quote
  expect(screen.getByTestId('quote')).toMatchSnapshot();
});
