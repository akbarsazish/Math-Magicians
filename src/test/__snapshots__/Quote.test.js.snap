// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qoutes component displays an error message on fetch failure 1`] = `
<div
  data-testid="errorState"
>
  Error: 
  Failed to fetch quote.
</div>
`;

exports[`Qoutes component displays the fetched quote correctly 1`] = `
<div
  class="quote"
  data-testid="quote"
>
  <p
    class="qouteText"
  >
     
    Mocked quote text
     - 
    Mocked author
     
  </p>
</div>
`;

exports[`Qoutes component displays the loading state 1`] = `
<div
  data-testid="loadingState"
>
  Loading...
</div>
`;
